src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,215 @@
+      .then((data) => pose = data)
+      .then(() => addMap());
+}
+
+/**
+ * Creates the 2D terrain and points a light at it.
+ * It also calls two other functions so it can wait on the fetch
+ * request and significantly cut down rendering time with
+ * geometry instancing.
+*/
+function addMap() {
+  // Creates the 2D map.
+  const loader = new THREE.TextureLoader();
+  let material = new THREE.MeshLambertMaterial({
+    map: loader.load(
+        'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng +
+      '&zoom=18&size=500x500&key=' + apiKey),
+  });
+  let geometry = new THREE.PlaneGeometry(50, 50);
+  const map = new THREE.Mesh(geometry, material);
+  map.position.set(0, -4, 0);
+  map.rotation.x = -Math.PI / 2;
+  scene.add(map);
+
+  // Add the light to the scene.
+  const light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 100, 0);
+  scene.add(light);
+  plotTrajectory();
+
+  // Adds pose data to the scene as one Geometry instance.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation);
+  plotOrientation();
+}
+
+/**
+ * Converts lla coordinates to local world coordinates.
+ * @param {double} lat Latitude in degrees.
+ * @param {double} lng Longitude in degrees.
+ * @param {double} alt Altitude in meters.
+ * @return {array} An array returning gps coordinates.
+ */
+function llaDegreeToLocal(lat, lng, alt) {
+  lng = (lng - pose[0].lng) * 25000 * poseTransform.scale;
+  alt = (alt - pose[0].alt)/4;
+  lat = (lat - pose[0].lat) * 25000 * -poseTransform.scale;
+  return [lat, alt, lng];
+}
+/**
+ * This uses the pose data to create a blue line representing the pose
+ * trajectory.
+ */
+function plotTrajectory() {
+  // Removes any existing trajectory objects for repositioning.
+  scene.remove(trajectory);
+  /**
+   * The x axis controls the left and right direction, the y axis controls
+   * up and down movement, and the z axis controls forward and back movement.
+   */
+  const coordinates=[];
+  for (point of pose) {
+    const localCoords = llaDegreeToLocal(point.lat, point.alt, point.lng);
+    /**
+      * Subtracting each point by the first point starts the pose
+      * at coordinates (0, 0, 0). Each unit in the 3D scene is 4
+      * meters while the 6th decimal point in GPS coordinates represents
+      * .11 meters. This is why we multiple the lat/lng by 25000, an
+      * increment in the 6th decimal place equates to a 0.025 unit
+      * change in our 3D space. Since altitude is already represented in
+      * meters, we simply divide by 4 to adjust for our 1:4 unit ratio.
+      */
+    coordinates.push(new THREE.Vector3(
+        localCoords[0],
+        localCoords[1],
+        localCoords[0]));
```

Comment:
I try to avoid indexing into an array to get specific meaningful elements like this. It's easy to mess up (and hard to review), because you have to keep 0=lat, etc in your head. Often the answer is to bundle into an object with named properties. Alternately, you could destructure into named variables:
```
const [x, y, z] = llaDegreeToLocal(lat, lng, alt);
Vector3(x, y ,z);
```
or spread the array into the vector constructor:
```
const localCoords = llaDegreeToLocal(lat, lng, alt);
Vector3(...localCoords);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax